src/components/common/Pagination/Pagination.jsx

```diff
@@ -0,0 +1,53 @@
+import { INFO_MESSAGES } from '@/constants/messages';
+import styles from './Pagination.module.scss';
+
+const Pagination = ({ currentPage, totalCount, pageSize, onPageChange }) => {
+  const totalPages = Math.ceil(totalCount / pageSize);
+  const groupSize = 5;
+  const currentGroup = Math.floor((currentPage - 1) / groupSize);
+  const startPage = currentGroup * groupSize + 1;
+  const endPage = Math.min(startPage + groupSize - 1, totalPages);
+
+  if (totalPages === 0) {
+    return <p>{INFO_MESSAGES.emptyList}</p>; // 나중에 UI 디자인 추가
+  }
+
+  const pageNumbers = Array.from(
+    { length: endPage - startPage + 1 },
+    (_, i) => startPage + i,
+  );
+
+  return (
+    <div className={styles.pagination}>
+      <button
+        onClick={() => onPageChange(currentPage - 1)}
+        disabled={currentPage === 1}
+        className={`${styles.arrow} ${styles.prev}`}
+        aria-label="Previous page"
+      />
+
+      {pageNumbers.map((page) => (
+        <button
+          key={page}
+          onClick={() => onPageChange(page)}
```

Comment:
💊 제안
버튼 태그로 잘 작업하셨어요! 이런 경우 아래처럼 이벤트 객체를 통해 접근하실 수도 있습니다.
개인적으로 코드도 깔끔해지고 이벤트 객체로 받는것이 더 적절하다고 생각해서 추천드려요!

```suggestion
        <button
          key={page}
          value={page}
          onClick={onPageChange} // onPageChange 함수에 event 자체를 인자로 받아 event.target.value 이렇게 접근
```